### PR TITLE
validate response of federation APIs

### PR DESCRIPTION
**Pull request by MFTabriz**
- verify json responses are parsable
- set `Accept` request header and check `Content-Type` in response headers
- refuse empty and invalid webfinger queries more strictly (+tests)
- log the reasons for webfinger query rejections in debug mode
- validate server URL before setting it through API (+test failed validations with `failAdminRequest()`)
- bump supertest minor version
- update `test/automated/api/package-lock.json`

### CODE REVIEW

File: controllers/admin/config.go [hunk 1]
```diff
@@ -416,7 +416,8 @@ func SetServerURL(w http.ResponseWriter, r *http.Request) {
 	}
 
 	rawValue, ok := configValue.Value.(string)
-	if !ok {
+	serverHostString := utils.GetHostnameFromURLString(rawValue)
```
Comment (gabek) [inline controllers/admin/config.go:424]: If `ok` is false then `rawValue` I'm assuming is an empty string. I'm not sure what happens when an empty string is passed to `GetHostnameFromURLString`.

Comment (MFTabriz) [reply controllers/admin/config.go:424]: it calls net/`url.Parse()` which returns an empty string and no error

Comment (MFTabriz) [reply controllers/admin/config.go:424]: I separated the checks just in case.

**Comment by gabek** [comment]
Thank you for adding all these!  :1st_place_medal:

**Pull request merged**